app/server/lib/OIDCConfig.ts

```diff
@@ -138,25 +194,23 @@ export class OIDCConfig {
     const mreq = req as RequestWithLogin;
     try {
       const params = this._client.callbackParams(req);
-      const { state, targetUrl } = mreq.session?.oidc ?? {};
-      if (!state) {
-        throw new Error('Login or logout failed to complete');
-      }
-
-      const codeVerifier = await this._retrieveCodeVerifierFromSession(req);
+      const { targetUrl } = mreq.session?.oidc ?? {};
+      const checks = await this._retrieveChecksFromSession(mreq);
 
       // The callback function will compare the state present in the params and the one we retrieved from the session.
       // If they don't match, it will throw an error.
-      const tokenSet = await this._client.callback(
-        this._redirectUrl,
-        params,
-        { state, code_verifier: codeVerifier }
-      );
+      const tokenSet = await this._client.callback(this._redirectUrl, params, checks);
```

Comment:
Thanks for your feedback! I'll take a look at this next week (I am not working for the ANCT today).

The last time I took a look at Authentik, I failed to setup an instance (at least, it seemed to me it's not *that* easy). I'll take a look and see if a [free account on their SAAS offer](https://goauthentik.io/pricing/) would be enough.

Reply:
I could reproduce the issue: one have to set `GRIST_OIDC_IDP_ENABLED_PROTECTIONS=PKCE`.

I achieved to fix that by sanitizing the parameters.

However, I am not very comfortable with that: 
- Keycloak seem to not pass any state at all in such case, so it seems more like we would work around a weird behavior of Authentik (and thus a bit of complexity for that);
- Authentik seems to support natively the STATE protection, so it would mean that the user has to explicitly remove that protection, which makes me think it would rather be an edge case;

I tend to think we may see if a user complains about this and fix that in such case. What do you think?

Reply:
Here is the patch I made, if we may want to apply the work around:

```patch
diff --git a/app/server/lib/OIDCConfig.ts b/app/server/lib/OIDCConfig.ts
index 1e3e9678..3e97fe42 100644
--- a/app/server/lib/OIDCConfig.ts
+++ b/app/server/lib/OIDCConfig.ts
@@ -64,6 +64,7 @@
 import * as express from 'express';
 import { GristLoginSystem, GristServer } from './GristServer';
 import {
+    CallbackParamsType,
   Client, ClientMetadata, generators, Issuer, TokenSet, UserinfoResponse
 } from 'openid-client';
 import { Sessions } from './Sessions';
@@ -277,7 +278,7 @@ export class OIDCConfig {
   public async handleCallback(sessions: Sessions, req: express.Request, res: express.Response): Promise<void> {
     try {
       const mreq = this._getRequestWithSession(req);
-      const params = this._client.callbackParams(req);
+      const params = this._sanitizeParams(this._client.callbackParams(req));
       const { targetUrl } = mreq.session.oidc ?? {};
 
       const checks = this._protectionManager.asObjForCallback(mreq.session.oidc ?? {});
@@ -388,6 +389,14 @@ export class OIDCConfig {
     });
   }
 
+  private _sanitizeParams(params: CallbackParamsType): CallbackParamsType {
+    return Object.fromEntries(
+      Object.entries(params).flatMap(([key, value]) => {
+        return value ? [[key, value]] : [];
+      })
+    );
+  }
+
   private _getRequestWithSession(req: express.Request) {
     const mreq = req as RequestWithLogin;
     if (!mreq.session) { throw new Error('no session available'); }
```